TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/PrettyPrinterTest.java

```diff
@@ -0,0 +1,115 @@
+package org.opensearch.migrations.replay;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opensearch.migrations.testutils.CountingNettyResourceLeakDetector;
+import org.opensearch.migrations.testutils.TestUtilities;
+import org.opensearch.migrations.testutils.TestWithNettyLeakDetection;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@TestWithNettyLeakDetection
+public class PrettyPrinterTest {
```

Comment:
Not blocking, Do you think there are other test cases to account for? an empty buffer or a buffer with headers but no body for example.

Reply:
Yes, I'm not sure what the behavior should be there, so tests would be a good exercise to find out and to codify.